x-pack/legacy/plugins/ml/public/jobs/new_job_new/pages/components/datafeed_step/components/query_delay/query_delay_input.tsx

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useState, useContext, useEffect } from 'react';
+import { EuiFieldText } from '@elastic/eui';
+import { JobCreatorContext } from '../../../job_creator_context';
+import { Description } from './description';
+
+const DEFAULT_QUERY_DELAY = '60s';
+
+export const QueryDelayInput: FC = () => {
+  const { jobCreator, jobCreatorUpdate, jobValidator, jobValidatorUpdated } = useContext(
+    JobCreatorContext
+  );
+  const [validation, setValidation] = useState(jobValidator.queryDelay);
+  const [queryDelay, setQueryDelay] = useState(
+    jobCreator.queryDelay === null ? '' : jobCreator.queryDelay
```

Comment:
Similar to the other comment, maybe `jobCreator.queryDelay === null ? '' : jobCreator.queryDelay` could go into a function and be reused.